tt_metal/CMakeLists.txt

```
@@ -63,6 +63,7 @@ set_target_properties(
             FALSE
 )
 
+file(GLOB_RECURSE COMPUTE_KERNEL_API include/*)
```

Do we know that everything under `include/` is device-side?  I'm not aware of any such assurance.
Device-side headers are required at runtime, while host-side headers are only required at build time.  So they go into distinct packages.

```suggestion
file(GLOB_RECURSE COMPUTE_KERNEL_API include/compute_kernel_api/*)
```

---

> Do we know that everything under include/ is device-side? I'm not aware of any such assurance.

I think we should make that an invariant ...

Because you introduced the lovely `api` folder for anything that is not device side.